manifests/rules.pp

```diff
@@ -1,8 +1,14 @@
 # apply user-defined rules
-class logrotate::rules ($rules = $::logrotate::rules){
+class logrotate::rules {
 
```

Comment:
Should we support direct `lookup()` calls here or should this be a Hiera auto-lookup?

Reply:
Not super sure. It seems like you would want to do a lookup but really the argument of either deep or first is probably what the user would want to configure.

Reply:
The hiera direct is just moved into the body now. If you specify a `::logratotate::hieramerge` it will use the lookup func with a deep merge which is what hiera_hash used to do. Else it will just use the direct lookup via puppets normal `::logroate::rules`

Reply:
@cwood I believe it would be best to keep our module configured to utilize hiera autolookup for hieradata lookups and have users utilize the `lookup()` function in their profiles if they prefer hard-coded lookups within their code.

For example a hiera automatic lookup that would work would look like so:
``` yaml
lookup_options:
  logrotate::rules:
    merge: deep
logrotate::rules:
  messages:
    path: '/var/log/messages'
    rotate: 5
    rotate_every: 'week'
    postrotate: '/usr/bin/killall -HUP syslogd'
```
^ This would automatically set the `$rules` parameter in the `logrotate` class to the above hash, which would then get passed to `logrotate::rule` via `logrotate::rules` without the need for hardcoding a `lookup()` call into the class.

Additionally, if they want to hardcode the lookup, they could do so this way:
``` puppet
class profile::logroate {
  $rules = lookup('logrotate::rules', Hash, 'deep')
  class { 'logrotate':
    rules => $rules,
  }
}
```

This gives our users the option to utilize the exposed `$rules` parameter in whatever way fits them best.

Reply:
Hey @dhollinger 

Get that these are all proper workarounds to this issue however as someone who was using this module at an older version then upgrading and the same behavior doesn't work was something that I was relying on.

I think that if this modules maintainers wanna move away from the `$hieramerge` option I should update this PR to remove it so peoples systems breaks and they know they need to update to use these said workarounds. 

Maybe a compromise is to just use the older `hiera_hash` which is what this module used to do and didn't break what I was sort of expecting. Although the `lookup` function update I proposed will also do the same thing and is the more modern way to do `hiera_hash`.

Just to loop back to the hiera lookup. That will still work the same in the way I put it here. Puppet will still do the lookup regardless if it is done in the body or in the arguments of the class. 

Let me know if you want me to update this PR to just remove the option or maybe we can compromise and keep the backwards compatibility along with maybe adding a warning that this option will be depreciated.